pkg/repo/oracle.go

```diff
@@ -81,30 +79,5 @@ func (d *oracleRepo) GetKernelPackages(
 
 	sort.Sort(pkg.ByVersion(pkgs)) // so kernels can be skipped if previous has BTF already
 
-	for i, pkg := range pkgs {
-		log.Printf("DEBUG: start pkg %s (%d/%d)\n", pkg, i+1, len(pkgs))
-
-		// Jobs about to be created:
-		//
-		// 1. Download package and extract vmlinux file
-		// 2. Extract BTF info from vmlinux file
-
-		err := processPackage(ctx, pkg, workDir, force, jobChan)
-		if err != nil {
-			if errors.Is(err, utils.ErrHasBTF) {
-				log.Printf("INFO: kernel %s has BTF already, skipping later kernels\n", pkg)
-				return nil
-			}
-			if errors.Is(err, context.Canceled) {
-				return nil
-			}
-
-			log.Printf("ERROR: %s: %s\n", pkg, err)
-			continue
-		}
-
-		log.Printf("DEBUG: end pkg %s (%d/%d)\n", pkg, i+1, len(pkgs))
-	}
-
-	return nil
+	return processPackages(ctx, workDir, pkgs, opts, chans)
```

Comment:
I see some repos use the common `processPackages` function but others not, shouldn't they all use the common function for the loop? Would also solve the part about the missing `!opts.Ordered` check.

Reply:
Let me take a look. Adding the `Ordered` thing was a bit of a hack to ensure we didn't miss some packages. Most of the time we don't want it because it results in unnecessary work.